src/Illuminate/Database/Connection.php

```diff
@@ -965,7 +965,11 @@ public function beforeExecuting(Closure $callback)
      */
     public function listen(Closure $callback)
     {
-        $this->events?->listen(Events\QueryExecuted::class, $callback);
+        $this->events?->listen(Events\QueryExecuted::class, function ($event) use ($callback) {
+            if ($event->connection === $this) {
+                $callback();
+            }
+        });
```

Comment:
This is an identity check. I felt this was better than checking the connection name.

If you change the driver instance, it feels like you would want to also remove the listeners.